### PR TITLE
Add unicode_normalize specs

### PR DESCRIPTION
**Pull request by jsyeo**
For #3264.

### CODE REVIEW

File: spec/ruby/core/string/unicode_normalize_spec.rb [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  end
+
+  it "defaults to the nfc normalization form if no forms are specified" do
+    @accented_f.unicode_normalize.should == "\u1e9b\u0323"
+    @angstrom.unicode_normalize.should == "\u00c5"
+    @ohm.unicode_normalize.should == "\u03a9"
+  end
+
+  it "raises an Encoding::CompatibilityError if the string is not in an unicode encoding" do
+    lambda { @angstrom.force_encoding("ISO-8859-1").unicode_normalize }.should raise_error(Encoding::CompatibilityError)
```
Comment (YorickPeterse) [inline spec/ruby/core/string/unicode_normalize_spec.rb]: Could you wrap this line at ~80 characters per line?

File: spec/ruby/core/string/unicode_normalize_spec.rb [hunk 2]
```diff
@@ -0,0 +1,97 @@
+require File.expand_path('../../../spec_helper', __FILE__)
+
+# Examples taken from http://www.unicode.org/reports/tr15/#Norm_Forms
+
+describe "String#unicode_normalize" do
+  before :each do
```
Comment (YorickPeterse) [inline spec/ruby/core/string/unicode_normalize_spec.rb]: `:each` isn't really needed since it's the default mode if I'm not mistaken.

**Comment by jsyeo** [comment]
@YorickPeterse I've updated the PR.

**Comment by YorickPeterse** [comment]
Thanks!

**Pull request merged**